src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size = 0;
+
+    private class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
```

Comment:
Diamond operator should be used properly to avoid raw type usage. Use `new Node<K,V>[DEFAULT_CAPACITY]` instead of `new Node[DEFAULT_CAPACITY]` to ensure type safety.

Reply:
please follow AI's advice

Reply:
![2024-08-13 08_00_48-jv-hash-map – MyHashMap java](https://github.com/user-attachments/assets/4a31b68b-f2bf-441e-a2a7-d8e188d08d3c)